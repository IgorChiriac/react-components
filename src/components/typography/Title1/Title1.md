```js
<Title1>Title1</Title1>
```

```js
<Title1 style={{ textAlign: 'center', color: 'salmon' }}>Title1</Title1>
```
