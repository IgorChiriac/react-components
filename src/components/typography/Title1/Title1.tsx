import React from 'react';
import styled from 'styled-components';
import * as fonts from '../../../constants/fonts';

export interface ITitle1Props extends React.HTMLAttributes<HTMLHeadingElement> {}

const STitle = styled.h1`
  font-weight: normal;
  font-size: 48px;
  line-height: 56px;
  letter-spacing: 0;
  font-family: ${fonts.openSans};
`;

const Title1 = (props: ITitle1Props) => {
  return <STitle {...props} />;
};

export default Title1;
