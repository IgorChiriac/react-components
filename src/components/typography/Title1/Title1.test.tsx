import { axe } from 'jest-axe';
import React from 'react';
import { render } from 'react-testing-library';
import Title1 from './Title1';

describe('<Title1 />', () => {
  it('renders the component with props with no a11y violations', async () => {
    const { container } = render(<Title1 style={{ textAlign: 'center' }}>test</Title1>);
    expect(container.firstChild).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
