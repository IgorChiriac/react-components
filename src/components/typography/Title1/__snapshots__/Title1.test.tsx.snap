// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title1 /> renders the component with props with no a11y violations 1`] = `
.c0 {
  font-weight: normal;
  font-size: 48px;
  line-height: 56px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
}

<h1
  class="c0"
  style="text-align: center;"
>
  test
</h1>
`;
